### PR TITLE
remove chartmuseum backend

### PR DESCRIPTION
**Pull request by wy65701436**
Harbor deprecates chartmuseum as of v2.8.0

Epic: https://github.com/goharbor/harbor/issues/17958 
Discussion: https://github.com/goharbor/harbor/discussions/15057

Signed-off-by: Wang Yan <wangyan@vmware.com>

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Comment by chlins** [comment]
The chart legacy APIs should also be cleaned from swagger.
- https://github.com/goharbor/harbor/blob/main/api/swagger.yaml
- https://github.com/goharbor/harbor/blob/main/api/v2.0/legacy_swagger.yaml

File: make/photon/prepare/migrations/version_1_10_0/harbor.yml.jinja [hunk 1]
```diff
@@ -287,7 +287,6 @@ external_redis:
   # db_index 0 is for core, it's unchangeable
   registry_db_index: {{ external_redis.registry_db_index }}
   jobservice_db_index: {{ external_redis.jobservice_db_index }}
-  chartmuseum_db_index: {{ external_redis.chartmuseum_db_index }}
```
Comment (chlins) [inline make/photon/prepare/migrations/version_1_10_0/harbor.yml.jinja:290]: Do we need to remove the chartmuseum related config in the old version template? Not sure about that.

**Comment by chlins** [comment]
Source code structure mentioned `chartserver` in the https://github.com/goharbor/harbor/blob/main/CONTRIBUTING.md.

**Review by chlins — APPROVED** [review]
lgtm

File: src/pkg/artifact/dao/model.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ func init() {
 // Artifact model in database
 type Artifact struct {
 	ID                int64     `orm:"pk;auto;column(id)"`
-	Type              string    `orm:"column(type)"`                // image or chart
```
Comment (stonezdj) [inline src/pkg/artifact/dao/model.go:33]: Pls keep the previous comments, the artifact is used for oci registry, not the chartmuseum artifact,  the type could be chart if it is a chart with OCI artifact.

File: src/pkg/artifact/model.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ import (
 // for all users.
 type Artifact struct {
 	ID                int64                  `json:"id"`
-	Type              string                 `json:"type"`                // image, chart, etc
```
Comment (stonezdj) [inline src/pkg/artifact/model.go:34]: Pls keep the previous comments, the artifact is used for oci registry, not the chartmuseum artifact,  the type could be chart if it is a chart with OCI artifact.

File: src/controller/artifact/processor/chart/chart.go [hunk 4]
```diff
@@ -16,17 +16,11 @@ package chart
 
 import (
 	"context"
-	"encoding/json"
```
Comment (stonezdj) [inline src/controller/artifact/processor/chart/chart.go:19]: Please keep this file unchanged, this code is used to handle OCI chart artifact.

**Comment by stonezdj** [comment]
> The chart legacy APIs should also be cleaned from swagger.
> 
> * https://github.com/goharbor/harbor/blob/main/api/swagger.yaml
> * https://github.com/goharbor/harbor/blob/main/api/v2.0/legacy_swagger.yaml

+1

File: make/photon/prepare/migrations/version_1_10_0/harbor.yml.jinja (hunk 1 above)
Comment (stonezdj) [reply make/photon/prepare/migrations/version_1_10_0/harbor.yml.jinja:290]: +1, they are history file, no need to change it.

File: src/pkg/artifact/model.go (hunk 3 above)
Comment (wy65701436) [reply src/pkg/artifact/model.go:34]: fixed

File: src/controller/artifact/processor/chart/chart.go (hunk 4 above)
Comment (wy65701436) [reply src/controller/artifact/processor/chart/chart.go]: fixed.

**Review by stonezdj — APPROVED** [review]
LGTM

**Pull request merged**